gobblin-metrics/src/main/java/gobblin/metrics/KafkaReporter.java

```diff
@@ -0,0 +1,421 @@
+    }
+
+    @Override
+    protected BuilderImpl self() {
+      return this;
+    }
+  }
+
+  /**
+   * Builder for {@link gobblin.metrics.KafkaReporter}
+   * Defaults to no filter, reporting rates in seconds and times in milliseconds
+   */
+  public static abstract class Builder<T extends Builder<T>> {
+    protected MetricRegistry _registry;
+    protected String _name;
+    protected MetricFilter _filter;
+    protected TimeUnit _rateUnit;
+    protected TimeUnit _durationUnit;
+    protected String _host;
+    protected String _env;
+    protected Set<String> _tags;
+
+    protected Builder(MetricRegistry registry) {
+      this._registry = registry;
+      this._name = "KafkaReporter";
+      this._rateUnit = TimeUnit.SECONDS;
+      this._durationUnit = TimeUnit.MILLISECONDS;
+      this._filter = MetricFilter.ALL;
+      try {
+        this._host = InetAddress.getLocalHost().getHostName();
+      } catch(UnknownHostException e) {
+        this._host = "";
+      }
+      this._env = "dev";
+      this._tags = new HashSet<String>();
+    }
+
+    protected abstract T self();
+
+    /**
+     * Only report metrics which match the given filter.
+     *
+     * @param filter a {@link MetricFilter}
+     * @return {@code this}
+     */
+    public T filter(MetricFilter filter) {
+      this._filter = filter;
+      return self();
+    }
+
+    /**
+     * Convert rates to the given time unit.
+     *
+     * @param rateUnit a unit of time
+     * @return {@code this}
+     */
+    public T convertRatesTo(TimeUnit rateUnit) {
+      this._rateUnit = rateUnit;
+      return self();
+    }
+
+    /**
+     * Convert durations to the given time unit.
+     *
+     * @param durationUnit a unit of time
+     * @return {@code this}
+     */
+    public T convertDurationsTo(TimeUnit durationUnit) {
+      this._durationUnit = durationUnit;
+      return self();
+    }
+
+    /**
+     * Set host
+     * @param host hostname
+     * @return {@code this}
+     */
+    public T withHost(String host) {
+      this._host = host;
+      return self();
+    }
+
+    /**
+     * Set environment
+     * @param env environment
+     * @return {@code this}
+     */
+    public T withEnv(String env) {
+      this._env = env;
+      return self();
+    }
+
+    /**
+     * Add tags
+     * @param tags
+     * @return {@code this}
+     */
+    public T withTags(String... tags) {
+      Collections.addAll(this._tags, tags);
+      return self();
+    }
+
+    /**
+     * Builds and returns {@link gobblin.metrics.KafkaReporter}
+     *
+     * @param brokers string of Kafka brokers
+     * @param topic topic to send metrics to
+     * @return KafkaReporter
+     */
+    public KafkaReporter build(String brokers, String topic) {
+      return new KafkaReporter(_registry, _name, _filter, _rateUnit, _durationUnit,
+          brokers, topic, _host, _env, _tags);
+    }
+
+  }
+
+  /**
+   * Class to contain a single metric
+   */
+  public static class Metric {
```

Comment:
Not a good practice to have a value class with all public members (some of them are mutable). Also we are not sure if and how this class will be used by any backend. For now, I think it's good to keep it protected or even private if you find you absolutely need this class.